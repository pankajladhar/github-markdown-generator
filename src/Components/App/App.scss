@import './../../styles/githubMarkdown';

.container {
  .editor-container {
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    width: 50%;
    height: auto;
    overflow: auto;
    font-size: 12px;
    box-shadow: -2px 2px 2px 2px rgba(0, 0, 0, .4);
    z-index: 99;
    padding: 16px;
    padding-top: 75px;

    .editor {
      width: 100%;
      min-height: 400px;
      display: block;

      &:focus {
        outline: none;
      }
    }
  }

  .result-container {
    background: lavender;
    position: fixed;
    top: 0;
    right: 0;
    left: 50%;
    bottom: 0;
    overflow: auto;
    padding: 10px;
    padding-left: 20px;
    display: flex;
    flex-direction: column;

    .btn-wrapper {
      display: flex;
      flex-direction: column;
      position: fixed;
      right: 10px;
      top: 20px;
    }

    .button {
      margin-left: auto;
      margin-bottom: 10px;

      &:disabled {
        cursor: not-allowed;
        background: #e8e6e6;
        color: #333;
      }
    }

    textarea {
      -ms-text-size-adjust: 100%;
      -webkit-text-size-adjust: 100%;
      font-size: 16px;
      line-height: 1;
      word-wrap: break-word;
      width: 100%;
      -webkit-appearance: none;
      height: 100%;
      background: none;
      border: none;
      line-height: 22px;

      &:focus {
        outline: none;
      }
    }
  }
}
