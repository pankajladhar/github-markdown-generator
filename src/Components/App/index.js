import React, { Component } from 'react';
//import PropTypes from 'prop-types';
import { ActionData } from './../../utils/ActionData';
import ActionBar from './../ActionBar';

import './App.scss';

const propTypes = {}
const defaultProps = {}

export default class App extends Component {
    constructor(props) {
        super(props);
        this.actions = ActionData;
    }
    render() {
        return (
            <div>
                <div className="container">
                    <div className="editor-container">
                        <ActionBar actions={this.actions}/>
                        <div id="editor" 
                            className="editor markdown-body" 
                            onInput={this.onGenerateClick} 
                            contentEditable="true" ref={(editor) => { this.textEditor = editor }}>
                        </div>
                    </div>
                    <div className="result-container">
                        <textarea readOnly className="editor1"></textarea>
                    </div>
                </div>
            </div>
        );
    }
}

App.propTypes = propTypes;
App.defaultProps = defaultProps;
