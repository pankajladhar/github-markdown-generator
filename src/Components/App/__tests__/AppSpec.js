import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import App from './../index';

describe('Button', ()=>{
    xit('renders correctly', ()=>{
        let tree = renderer.create(
            <App  />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
