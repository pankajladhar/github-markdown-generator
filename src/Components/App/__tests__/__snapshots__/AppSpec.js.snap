// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="editor-container"
    >
      <div
        className="actionbar-container"
      >
        <input
          className="button action"
          name={undefined}
          onClick={[Function]}
          title="Heading 1"
          type="button"
          value="H1"
        />
        <input
          className="button action"
          name={undefined}
          onClick={[Function]}
          title="Heading 2"
          type="button"
          value="H2"
        />
        <input
          className="button action"
          name={undefined}
          onClick={[Function]}
          title="Heading 3"
          type="button"
          value="H3"
        />
        <input
          className="button action"
          name={undefined}
          onClick={[Function]}
          title="Bold"
          type="button"
          value="B"
        />
        <input
          className="button action"
          name={undefined}
          onClick={[Function]}
          title="Italic"
          type="button"
          value="I"
        />
        <input
          className="button action"
          name={undefined}
          onClick={[Function]}
          title="Unordered List"
          type="button"
          value="UL"
        />
        <input
          className="button action"
          name={undefined}
          onClick={[Function]}
          title="Ordered List"
          type="button"
          value="OL"
        />
      </div>
      <div
        className="editor markdown-body"
        contentEditable="true"
        id="editor"
        onInput={undefined}
      />
    </div>
    <div
      className="result-container"
    >
      <textarea
        className="editor1"
        readOnly={true}
      />
    </div>
  </div>
</div>
`;
